src/libaktualizr/uptane/uptane_key_test.cc

```diff
@@ -57,49 +57,54 @@ void initKeyTests(Config& config, Uptane::SecondaryConfig& ecu_config1, Uptane::
   config.uptane.secondary_configs.push_back(ecu_config2);
 }
 
-void checkKeyTests(std::shared_ptr<INvStorage>& storage, std::shared_ptr<SotaUptaneClient> sota_client) {
-  std::string ca;
-  std::string cert;
-  std::string pkey;
-  EXPECT_TRUE(storage->loadTlsCreds(&ca, &cert, &pkey));
-  EXPECT_TRUE(ca.size() > 0);
-  EXPECT_TRUE(cert.size() > 0);
-  EXPECT_TRUE(pkey.size() > 0);
-
-  std::string primary_public;
-  std::string primary_private;
-  EXPECT_TRUE(storage->loadPrimaryKeys(&primary_public, &primary_private));
-  EXPECT_TRUE(primary_public.size() > 0);
-  EXPECT_TRUE(primary_private.size() > 0);
-
-  EcuSerials ecu_serials;
-  EXPECT_TRUE(storage->loadEcuSerials(&ecu_serials));
-  EXPECT_EQ(ecu_serials.size(), 3);
-
-  std::vector<std::string> public_keys;
-  std::vector<std::string> private_keys;
-  std::map<Uptane::EcuSerial, std::shared_ptr<Uptane::SecondaryInterface> >::iterator it;
-  for (it = sota_client->secondaries.begin(); it != sota_client->secondaries.end(); it++) {
-    if (it->second->sconfig.secondary_type != Uptane::SecondaryType::kVirtual &&
-        it->second->sconfig.secondary_type != Uptane::SecondaryType::kLegacy) {
-      continue;
+// This is a class solely for the purpose of being a FRIEND_TEST to
```

Comment:
A bit annoying but fine I guess

Reply:
Agreed, but what is the superior alternative?

Reply:
Haven't found it :), let's live with it

Reply:
If it makes you feel better (it won't), I had to already do the same type of thing with `uptane_test_vectors.cc`.